src/components/swap/PendingModalContent/index.tsx

```diff
@@ -223,14 +226,17 @@ export function PendingModalContent({
 }: PendingModalContentProps) {
   const { chainId } = useWeb3React()
 
-  const classicSwapConfirmed = useIsTransactionConfirmed(
-    swapResult?.type === TradeFillType.Classic ? swapResult.response.hash : undefined
-  )
+  const transaction = useTransaction(swapResult?.type === TradeFillType.Classic ? swapResult.response.hash : undefined)
+  const tokens = useAllTokensMultichain()
+
+  const swapActivity = transaction && chainId ? transactionToActivity(transaction, chainId, tokens) : undefined
+
+  const classicSwapConfirmed = swapActivity?.status === TransactionStatus.Confirmed
```

Comment:
this change ensures we update to the success state at the same time as other places in the app